### PR TITLE
change mem control lock usage

### PR DESCRIPTION
**Pull request by HTHou**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/exception/WriteProcessRejectException.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package org.apache.iotdb.db.exception;
+
+import org.apache.iotdb.rpc.TSStatusCode;
+
+public class WriteProcessRejectException extends WriteProcessException {
+
+  private static final long serialVersionUID = -4217324287547595610L;
+
+  public WriteProcessRejectException(String message) {
+    super(message, TSStatusCode.WRITE_PROCESS_ERROR.getStatusCode());
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/exception/WriteProcessRejectException.java:28]: add a new code?

**Review by jixuan1989 — CHANGES_REQUESTED** [review]
why not add a new error code for write_reject_because_overload?

File: server/src/main/java/org/apache/iotdb/db/exception/WriteProcessRejectException.java (hunk 1 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/exception/WriteProcessRejectException.java:28]: Good Idea!

**Review by qiaojialin — APPROVED** [review]

**Review by jixuan1989 — APPROVED** [review]

**Comment by jixuan1989** [comment]
Is this related to some known bug? If so, we need to cherry-pick to rel/0.11, otherwise  please delete `mem_control_op` branch

**Pull request merged**